model/src/main/kotlin/Package.kt

```diff
@@ -192,7 +192,8 @@ data class Package(
             vcs = vcsProcessed.takeIf { it != other.vcsProcessed }?.toCuration(),
             isMetadataOnly = isMetadataOnly.takeIf { it != other.isMetadataOnly },
             isModified = isModified.takeIf { it != other.isModified },
-            sourceCodeOrigins = sourceCodeOrigins.takeIf { it != other.sourceCodeOrigins }
+            sourceCodeOrigins = sourceCodeOrigins.takeIf { it != other.sourceCodeOrigins },
+            labels = labels.takeIf { it != other.labels }.orEmpty()
```

Comment:
The function docs say "All equal values are set to null", so I believe `orEmpty()` needs to be dropped here. Also see my comment below.

Reply:
BTW, as an unrelated observation, this function seems to miss quite a few `PackageCurationData` properties, like `purl`, `cpe`, ...

Reply:
> The function docs say "All equal values are set to null", so I believe `orEmpty()` needs to be dropped here. Also see my comment below.

The `map` is non-nullable, similar to the only other `map`, `declaredLicenseMapping`.

Reply:
I changed the implementation to how I understood your proposal.